app/views/unattended/provisioning_templates/host_init_config/host_init_config_default.erb

```diff
@@ -73,7 +73,7 @@ if command -v subscription-manager &>/dev/null; then
   subscription-manager refresh
 fi
 
-<% if host_param_true?('redhat_install_host_tools') -%>
+<% if host_param_true?('redhat_install_host_tools', true) -%>
```

Comment:
Since only debian, suse, and ubuntu need this by default, what do you think about adding specific checks here for those OSs to install katello host tools by default?
It might also be worth adding a check for `plugin_present?('katello')` since katello-host-tools isn't useful without Katello.

This template seems to make different assumptions compared to `redhat_register.erb` about the Foreman configuration.

Reply:
Also, is it safe to assume that katello-host-tools will be available in the host's repository?
I'm not sure exactly where this template comes into play, so I'm not sure if that's a safe assumption.

Reply:
The check about is plugin Katello used, is already there some lines above. 

I changed the default to the same as on redhat_register. Katello host tools should be delivered as this is done after the sub man registration.

Reply:
I think this is global registration where `redhat_register` is provisioning.

Reply:
> Also, is it safe to assume that katello-host-tools will be available in the host's repository?
> I'm not sure exactly where this template comes into play, so I'm not sure if that's a safe assumption.

If this template is executed after `subscription-manager register` has completed, the host should have available all repositories in its assigned content view environments. If not, it may not. (I don't know enough about Foreman provisioning to know which is the case.)

However, since the next line is installing Tracer, it'd have the same dependency there as well. So I think this is a reasonable place for it.

Reply:
Oh, and regarding the needed repositories not being available - we've had that issue in several places in the UI, and always settled on "you're responsible for getting the content into the right content view and enabling the appropriate repos." There simply isn't a way to automate that that makes sense for all situations.

Reply:
Sure, just set it in one of these eight places! :D https://docs.theforeman.org/nightly/Provisioning_Hosts/index-katello.html#Host_Parameter_Hierarchy_provisioning

I don't see that specific parameter documented anywhere. Would need some more input from provisioning folks.

Reply:
@maximiliankolb any thoughts regarding documentation? Can we add both parameter to the documentation (redhat_install_host_tools and redhat_install_host_tracer_tools)

Reply:
From a user perspective, I expect that provisioning a host or registering a host to Foreman+Katello means that I have access to package lists in Foreman UI and to hints if services/the Kernel has to be restarted. With this in mind, I'd recommend, _if at all_, documenting how to disable this by changing a default parameter.

Regardless of the OS, if you provision a host and register a host to Foreman+Katello, the templates should install whatever is necessary to have the best user experience. I cannot speak much for Foreman on EL/Deb. 

Current docs: [Installing Tracer](https://docs.theforeman.org/nightly/Managing_Hosts/index-katello.html#Installing_Tracer_managing-hosts) + [Enabling Tracer](https://docs.theforeman.org/nightly/Managing_Hosts/index-katello.html#enabling-tracer-on-a-host_managing-hosts)

If we change the default from "is not installed" to "is installed automatically because of a new default host param", I can make the appropriate changes in foreman-documentation.

Reply:
> Current docs: [Installing Tracer](https://docs.theforeman.org/nightly/Managing_Hosts/index-katello.html#Installing_Tracer_managing-hosts) + [Enabling Tracer](https://docs.theforeman.org/nightly/Managing_Hosts/index-katello.html#enabling-tracer-on-a-host_managing-hosts)

the important part being the Prerequisite, which you have to do yourself :)

![image](https://github.com/user-attachments/assets/f4e70c12-55c9-492a-943a-4c33ad4793db)

Reply:
But there's a difference in doing this _after_ your host is registered to F+K and you can provide content etc. vs registering a host using global host registration where you have to provide _just the right_ repos. Any more packages other than `subscription-manager` have again _more_ dependencies that users have to cater for.

Strictly speaking, the Client repo is not sufficient for any OS (I did not test RHEL):

````
$ podman run --rm -it almalinux:9.4

# rm -fr /etc/yum.repos.d/almalinux-*
# vi /etc/yum.repos.d/client.repo
# dnf makecache
...
# cat /etc/yum.repos.d/client.repo
[client]
baseurl=https://yum.theforeman.org/client/latest/el9/x86_64/
enabled=1
gpgcheck=0
name=client
sslverify=0

# dnf install katello-host-tools-tracer
Last metadata expiration check: 0:00:26 ago on Thu Oct 31 09:00:21 2024.
Error: 
 Problem: conflicting requests
  - nothing provides crontabs needed by katello-host-tools-tracer-4.4.0-1.el9.noarch from client
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
````

I assume that most packages in Client repos have the same issue, e.g. OpenSCAP client requires Ruby ...

My **conclusion**: Limit pkg installation to a minimum when provisioning/registering a host. After the host is registered to F+K with an AK, have the user provide the "default" OS repos (e.g. BaseOS + AppStream on EL8+) and then install whatever from the combination of default OS repos + Client repo.

**Disclaimer**: tested with a container. I do not have access to VMs that are not provisioned in an automated way and therefore have automated pkg installations ... One could argue that `crontab` is installed by default in a EL9 VM compared to a container.

Reply:
> - nothing provides crontabs needed by katello-host-tools-tracer-4.4.0-1.el9.noarch from client

This is in baseos

> My **conclusion**: Limit pkg installation to a minimum when provisioning/registering a host. After the host is registered to F+K with an AK, have the user provide the "default" OS repos (e.g. BaseOS + AppStream on EL8+) and then install whatever from the combination of default OS repos + Client repo.

IMHO you always want the "default" OS repos and I can't think of a use case where you don't provide those. When you're subscribed to a content view, you always want to include those. The way I always interpreted the note as that you want base + Foreman Client, not "just Foreman Client". The AK should provide access to both.